### PR TITLE
[cc] Add cc toolchain to starlark cc_proto_library

### PR DESCRIPTION
**Pull request by Yannic**
Fixes https://github.com/bazelbuild/bazel/issues/16577

### CODE REVIEW

**Comment by Yannic** [comment]
cc @buildbreaker2021 @meteorcloudy

File: src/main/starlark/builtins_bzl/common/cc/cc_proto_library.bzl [hunk 1]
```diff
@@ -293,4 +293,5 @@ cc_proto_library = rule(
             allow_files = False,
         ),
     },
+    toolchains = cc_helper.use_cpp_toolchain(),
```
Comment (fmeum) [inline src/main/starlark/builtins_bzl/common/cc/cc_proto_library.bzl:296]: Could you move this to `_cc_proto_aspect`? That should solve the test failure.

Comment (Yannic) [reply src/main/starlark/builtins_bzl/common/cc/cc_proto_library.bzl]: Thank you! Of course 🤦‍♀️

**Comment by fmeum** [comment]
@buildbreaker2021 Could you review? This has the fix for the fix as well as an integration test.

**Review by buildbreaker2021 — APPROVED** [review]

**Pull request closed without merging**